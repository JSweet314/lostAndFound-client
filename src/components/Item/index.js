import React from 'react';
import PropTypes from 'prop-types';

export const Item = (
  { itemId, status, name, description, date, handleOnClick }
) => (
  <article
    onClick={() => handleOnClick(itemId)}
    name={itemId}
    className='user-item'>
    <h3 name={itemId}>{status}: {name}</h3>
    <p name={itemId}>{description}</p>
    <p name={itemId}>{date}</p>
  </article>
);

Item.propTypes = {
  itemId: PropTypes.number.isRequired,
  status: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  handleOnClick: PropTypes.func.isRequired
};

export default Item;