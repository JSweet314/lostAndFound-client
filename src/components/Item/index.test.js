import React from 'react';
import { shallow } from 'enzyme';
import Item from './index';

describe('Item', () => {
  const mockHandleOnClick = jest.fn();
  it('should match a snapshot', () => {
    const wrapper = shallow(
      <Item 
        name='keys'
        itemId={1}
        status='lost'
        description='car keys'
        date='2018-04-04'
        handleOnClick={mockHandleOnClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});