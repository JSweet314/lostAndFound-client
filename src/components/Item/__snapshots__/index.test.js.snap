// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    date="2018-04-04"
    description="car keys"
    handleOnClick={[Function]}
    itemId={1}
    name="keys"
    status="lost"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          name={1}
>
          lost
          : 
          keys
</h3>,
        <p
          name={1}
>
          car keys
</p>,
        <p
          name={1}
>
          2018-04-04
</p>,
      ],
      "className": "user-item",
      "name": 1,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "lost",
            ": ",
            "keys",
          ],
          "name": 1,
        },
        "ref": null,
        "rendered": Array [
          "lost",
          ": ",
          "keys",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "car keys",
          "name": 1,
        },
        "ref": null,
        "rendered": "car keys",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-04-04",
          "name": 1,
        },
        "ref": null,
        "rendered": "2018-04-04",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            name={1}
>
            lost
            : 
            keys
</h3>,
          <p
            name={1}
>
            car keys
</p>,
          <p
            name={1}
>
            2018-04-04
</p>,
        ],
        "className": "user-item",
        "name": 1,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "lost",
              ": ",
              "keys",
            ],
            "name": 1,
          },
          "ref": null,
          "rendered": Array [
            "lost",
            ": ",
            "keys",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "car keys",
            "name": 1,
          },
          "ref": null,
          "rendered": "car keys",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-04-04",
            "name": 1,
          },
          "ref": null,
          "rendered": "2018-04-04",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
